syncserver/src/tokenserver/extractors.rs

```diff
@@ -159,26 +151,6 @@ impl TokenserverRequest {
             });
         }
 
-        // If there's no keys_changed_at on the request, there must be no value stored on the user
-        // record. Note that this is only relevant for BrowserID requests, since OAuth requests
-        // must always include a keys_changed_at header. The Python Tokenserver converts a NULL
-        // keys_changed_at on the user record to 0 in memory, which means that NULL
-        // keys_changed_ats are treated equivalently to 0 keys_changed_ats. This would allow users
-        // with a 0 keys_changed_at on their user record to hold off on sending a keys_changed_at
-        // in requests even though the value in the database is non-NULL. To be thorough, we
-        // handle this case here.
-        if auth_keys_changed_at.is_none()
```

Comment:
..whereas this case seems to definitely be true, `keys_changed_at` is enforced for oauth via the `KeyId` struct, ensuring one exists and is valid and is converted to i64.

Reply:
Ah, right! I got thrown by the comment. I'll add this back and remove the browserID stuff.

Reply:
Sorry, what I meant was I agreed with removing this section, I was just being verbose about how I audited the extractor code and it matched what the comment said: that oauth will always supply a `keys_changed_at` value. So let's nuke it!

Reply:
Hrm. So, let me counter.
As I understand this step currently, it's a counter for a record being potentially out of step for unknown reasons.
The modified code checks to see if there's an `auth_keys_changed_at` and then checks to see if the user record may have a `0` due to a previously incorrectly written record.  I think that may still be a valid thing to test for.

Reply:
Just to clarify: the code checks if there's *not* a `keys_changed_at` in the request, which the original comment suggests never happens on oauth, but that's not quite correct. Per the new, more accurate comment added: a `0` value is treated as `None` even on oauth. 👍